### PR TITLE
Fix meta package symlinks

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What

Update the typscript importer transformer links to point to the index file and not the src folder

# Why
Packages were hitting the index.ts not the public-index.ts (so a new packlet introduced in #4665 caused the non-public files to be pulled into the meta package)

# How Tested
CI should test this.

### CODE REVIEW

**Review by dmceachernmsft — APPROVED** [review]

**Review by dmceachernmsft — APPROVED** [review]

File: packages/communication-react/tsconfig.json [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
     "paths": {
-      "@internal/chat-component-bindings": ["../../chat-component-bindings/src"],
-      "@internal/calling-component-bindings": ["../../calling-component-bindings/src"],
-      "@internal/acs-ui-common": ["../../acs-ui-common/src"],
-      "@internal/calling-stateful-client": ["../../calling-stateful-client/src"],
-      "@internal/chat-stateful-client": ["../../chat-stateful-client/src"],
-      "@internal/react-components": ["../../react-components/src"],
-      "@internal/react-composites": ["../../react-composites/src"]
+      "@internal/chat-component-bindings": ["../../chat-component-bindings/src/index"],
+      "@internal/calling-component-bindings": ["../../calling-component-bindings/src/index-public"],
```
Comment (dmceachernmsft) [inline packages/communication-react/tsconfig.json:8]: this one should just be `index` not `index-public`

**Review by dmceachernmsft — APPROVED** [review]

File: packages/communication-react/tsconfig.json (hunk 1 above)
Comment (JamesBurnside) [reply packages/communication-react/tsconfig.json:8]: Why do you say that?

Comment (JamesBurnside) [reply packages/communication-react/tsconfig.json:8]: Ah yep `@internal/calling-stateful-client` has the public index not the `@internal/calling-component-bindings`

Comment (JamesBurnside) [reply packages/communication-react/tsconfig.json]: Updated

Comment (JamesBurnside) [reply packages/communication-react/tsconfig.json]: OK, reverted that packlet back to just `index` and not `index-public`, there's not the correct re-exports in the meta package -- would need to be its own PR

**Pull request merged**